addons/product/product.py

```diff
@@ -470,6 +470,27 @@ def _get_product_variant_count(self, cr, uid, ids, name, arg, context=None):
             res[product.id] = len(product.product_variant_ids)
         return res
 
+    def _compute_product_template_field(self, cr, uid, ids, names, arg, context=None):
+        ''' Compute the field from the product_variant if there is only one variant, otherwise returns 0.0 '''
+        import pdb; pdb.set_trace()
+        res = {id: {} for id in ids}
+        templates = [template for template in self.browse(cr, uid, ids, context=context)]
+        unique_templates = [template.id for template in templates if template.product_variant_count == 1]
+        for template in templates:
+            for name in names:
+                res[template.id][name] = getattr(template.product_variant_ids[0], name) if template.id in unique_templates else 0.0
+        return res     
+
+    def _set_product_template_field(self, cr, uid, product_tmpl_id, name, value, args, context=None):
+        ''' Set the standard price modification on the variant if there is only one variant '''
+        if context is None:
```

Comment:
As you are not actively using the context, it is not necessary to check its existence.